ang/civicase/Utils.js

```diff
@@ -553,48 +550,94 @@
     };
   });
 
-  angular.module('civicase').directive('dropdownToggle', function($timeout) {
+  angular.module('civicase').directive('dropdownToggle', function ($timeout) {
     return {
       restrict: 'C',
-      link: function(scope, $el, attrs) {
-        function dropDownFixPosition(button,dropdown){
-          var dropDownTop = - $(window).scrollTop() + button.offset().top + button.outerHeight();
-          dropdown.css('top', dropDownTop + "px");
-          if(dropdown.hasClass('dropdown-menu-right')) {
-            dropdown.css('left', button.offset().left - dropdown.outerWidth() + button.outerWidth() + "px");
-          }
-          else {
-            dropdown.css('left', button.offset().left + "px");
+      link: function (scope, $el, attrs) {
+        function dropDownFixPosition (button, dropdown) {
+          var dropDownTop = -$(window).scrollTop() + button.offset().top + button.outerHeight();
+          dropdown.css('top', dropDownTop + 'px');
+          if (dropdown.hasClass('dropdown-menu-right')) {
+            dropdown.css('left', button.offset().left - dropdown.outerWidth() + button.outerWidth() + 'px');
+          } else {
+            dropdown.css('left', button.offset().left + 'px');
           }
         }
 
-        $timeout(function() {
-          $($el).click(function(){
-            dropDownFixPosition($(this),$(this).next('.dropdown-menu'));
+        $timeout(function () {
+          $($el).click(function () {
+            dropDownFixPosition($(this), $(this).next('.dropdown-menu'));
           });
 
-          document.addEventListener('scroll', function(e){
-            dropDownFixPosition($($el),$($el).next('.dropdown-menu'));
+          document.addEventListener('scroll', function (e) {
+            dropDownFixPosition($($el), $($el).next('.dropdown-menu'));
           }, true);
         });
       }
     };
   });
 
-  angular.module('civicase').directive('calculateScrollWidth', function() {
+  angular.module('civicase').directive('calculateScrollWidth', function () {
     return {
       restrict: 'A',
-      link: function(scope, $el, attrs) {
+      link: function (scope, $el, attrs) {
         scope.$watch(
           function () {
             return $('.case-list-table thead.affix-top').css('width');
           },
-          function(width) {
+          function (width) {
             $('.custom-scroll').css('width', width);
           }
         );
       }
     };
   });
 
+  /**
+   * Directive for the sticky table header functionality on case list page
+   */
+
+  angular.module('civicase').directive('stickyTableHeader', function () {
+    return {
+      restrict: 'A',
+      link: function (scope, $el, attrs) {
+        $(window).scroll(function () {
```

Comment:
Use `$window` instead.

Reply:
Also, should not you check for the scrolling inside the table only?

Reply:
`$window` will be undefined. We will have to use `$(window)` which is actually jQuery interpretation of window object.

Reply:
>Also, should not you check for the scrolling inside the table only?

The pager needs to sticky only when it is not in view on window. So the logic should still work if the table to fully scrolled and hence we need to hook into `$(window).scroll`.

Reply:
> $window will be undefined. We will have to use $(window) which is actually jQuery interpretation of window object.

`$window` is a dependency that needs to be injected. Otherwise it will be undefined.